reaper.go

```diff
@@ -117,6 +118,13 @@ func lookUpReaperContainer(ctx context.Context, sessionID string) (*DockerContai
 			return err
 		}
 
+		// if a health status is present on the container, and the container is not healthy, error
+		if r.healthStatus != "" {
+			if r.healthStatus != types.Healthy && r.healthStatus != types.NoHealthcheck {
+				return fmt.Errorf("container %s is not healthy, wanted status=%s, got status=%s", resp[0].ID[:8], types.Healthy, r.healthStatus)
+			}
+		}
+
```

Comment:
Do you think we could add this check as a readiness hook? Then I think this fix would be more intentional. Thoughts?

Reply:
In fact, we do have a [wait.ForHealth](https://golang.testcontainers.org/features/wait/health/), that could be of our interest 🤔 Wdyt?